### ISSUE TITLE
azurerm_mysql_database resource wants to re-create on every apply

### ISSUE THREAD

**Issue by scarolan** [terraform-providers/terraform-provider-azurerm#1183]
If you build an azurerm_mysql_database using the settings from the example code:

```
  charset             = "utf8"
  collation           = "utf8_unicode_ci"
```

And then do a subsequent `terraform plan` this is what happens:

```
Terraform will perform the following actions:

-/+ azurerm_mysql_database.employees (new resource required)
      id:                  "/subscriptions/c0a607b2-6372-4ef3-abdb-dbe52a7b56ba/resourceGroups/SeanC-Azure-Vault-SQL-Demo/providers/Microsoft.DBforMySQL/servers/mysql-server-1/databases/employees" => <computed> (forces new resource)
      charset:             "latin1" => "utf8" (forces new resource)
      collation:           "latin1_swedish_ci" => "utf8_unicode_ci" (forces new resource)
```

I have no idea where Latin Swedish is coming from.  I never set that anywhere.

**Comment by tombuildsstuff** [comment]
hey @scarolan 

Thanks for opening this issue :)

I have a feeling this will be fixed by #1154 (since this looks to be an Azure API bug) - which we're working through at the moment. In the interim you should be able to use [the `ignore_changes` attribute](https://www.terraform.io/docs/configuration/resources.html#ignore_changes) to workaround this issue.

Thanks!

**Comment by jeffreyCline** [comment]
@scarolan @tombuildsstuff 

I've looked into this and it is a casing issue with the API and the HCL file. I've created a private build where I suppress the case diff and it works as expected. I will send out a PR shortly with the fix. While I'm in the code I am also going to validate if MySQL has the same collation issues as PostgreSQL (issues #396 and #760 )

**Comment by jeffreyCline** [comment]
MySQL does not have the same collation issue as PostgreSQL, the collation issue above:

```bash
collation:           "latin1_swedish_ci" => "utf8_unicode_ci" (forces new resource)
```

Was fixed with the `2017-12-01 GA API` as @tombuildsstuff  had suspected. The API now always returns lowercased charset, so I added a ignore case to the schema for the database resource which fixes the issue.

**Comment by katbyte** [comment]
Hey @scarolan ,

Just wanted to let you know we have released [v1.6.0](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/CHANGELOG.md#160-may-24-2018) of the provider fixing this collation issue 🙂
